rust_icu_uloc/src/lib.rs

```diff
@@ -229,6 +233,58 @@ impl ULoc {
         )
     }
 
+    /// Implements `uloc_openKeywords()` from ICU4C.
+    pub fn keywords(&self) -> impl Iterator<Item = String> {
+        rust_icu_uenum::uloc_open_keywords(&self.repr)
+            .unwrap()
+            .map(|result| result.unwrap())
+    }
+
+    /// Implements `icu::Locale::getUnicodeKeywords()` from the C++ API.
+    pub fn unicode_keywords(&self) -> impl Iterator<Item = String> {
+        self.keywords().filter_map(|s| to_unicode_locale_key(&s))
+    }
+
+    /// Implements `uloc_getKeywordValue()` from ICU4C.
+    pub fn keyword_value(&self, keyword: &str) -> Result<Option<String>, common::Error> {
```

Comment:
I've been thinking... does this need to return `common::Error`?

Looks like, `keyword` could have an interior `NUL` byte technically speaking.  Could we, instead, use the potential `NulError` to re-attempt the conversion up to and including the said `NUL`?  Looks like it will make the type signature of the return value be just `Option<String>` which is nice, and I don't see that we lose anything.

Are you aware of any hidden perils of this approach?  Seems like such an approach could simplify a lot of the code below?

Also, sorry for not noticing this before --- seems like we shouldn't need to support `NUL` bytes in keywords, as the underlying ICU4C wouldn't know what to do with it anyways.  This makes it unnecessary to worry about `common::Error::string_with_interior_nul`.  `NulError` allows to [find the `NUL` byte position](https://doc.rust-lang.org/std/ffi/struct.NulError.html#method.nul_position), so perhaps we can re-attempt the conversion with this extra bit of information if it fails with `NulError` the first time around.

Thoughts?

Reply:
I'd be okay with reattempting the `CString` conversions when there's a `NulError`.

However, in this case `uloc_getKeywordValue` can also return an error if the keyword name is ill-formed, which is a case we'd want to handle.

Reply:
Aren't we swallowing that error by calling `.map(...)` in line 267?  

What am I missing?  The only place where I see an error can be returned here is `.map_err(...)` in line 261.

Reply:
Oh, I guess the `.map` will leave the possible `Error` in. OK.